src/sentry/models/project.py

```diff
@@ -384,3 +386,20 @@ def get_security_token(self):
 
     def get_lock_key(self):
         return 'project_token:%s' % self.id
+
+    def record_redirect_slug(self, historic_slug):
+        """
+        Records a historic slug used for redirect purposes. Overwrites
+        historic redirect slugs from previous projects
+        """
+        from sentry.models import ProjectRedirect
+
+        redirect, created = ProjectRedirect.objects.get_or_create(
+            redirect_slug=historic_slug,
+            organization=self.organization,
+            defaults={'project': self},
+        )
+
+        if not created:
+            redirect.project = self
```

Comment:
nit: dont use save() as it squashes the entire object

``redirect.update(project=self)``

Reply:
dc3a510